src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -98,10 +128,24 @@ trait RegularSync {
 
     parentByNumber match {
       case Some(parent) if checkHeaders(headers) =>
-        //we have same chain
+        //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
-          val hashes = headersQueue.take(blockBodiesPerRequest).map(_.hash)
-          waitingForActor = Some(context.actorOf(SyncBlockBodiesRequestHandler.props(peer, hashes)))
+
+          val currentBranchTotalDifficulty = headersQueue.map(_.number)
+            .map(blockNumber => blockchain.getBlockByNumber(blockNumber))
+            .collect{
+              case Some(b) => b.header.difficulty
+              case None => BigInt(0)
+            }.sum
+
+          val newBranchTotalDifficulty = headersQueue.map(_.difficulty).sum
+
+          if (currentBranchTotalDifficulty < newBranchTotalDifficulty) {
+            val hashes = headersQueue.take(blockBodiesPerRequest).map(_.hash)
+            waitingForActor = Some(context.actorOf(SyncBlockBodiesRequestHandler.props(peer, hashes)))
+          } else {
+            scheduleResume()
+          }
```

Comment:
this is for calculation of total difficulty of branch that we currently have vs branch that we are getting from peer, if ours is "heavier" then we are are ignoring branch from peer and sheduling resume if peers branch is "heavier" then we are asking for block bodies and start replacing ours branch with branch from peer